[#6] OwnerShopService 구현

tncks93 opened the pull request:
## 변동 사항
- OwnerShopService를 구현하였고, 그에 따라 OwnerShopRepository코드를 약간 수정했습니다.

## 문제점
- OwnerShopService에서 OwnerShopRepository와 OwnerShopRepository를 사용하고 있는데, 현재 OwnerShopRepository의
내부는 구현이 되어있지 않습니다. 슬랙에 보내드린 것 같이 Owner에 대한 정보를 어떻게 처리할지 아직 해결이 안되었습니다. 
- 따라서 ArgumentResolver구현과 Controller 수정은 하위 브랜치로 PR 올리겠습니다.

tncks93 committed:
[#6] Refactor: DTO 필드 추가 및 변수명 변경

tncks93 committed:
[#6] Feat: Owner용 Shop Service 구현

- 가게등록,조회,수정,open/close 상태 변경 서비스 구현
- 서비스 구현에 따라 OwnerShopRepository 코드 수정

tncks93 committed:
[#6] Test: OwnerShopRepository의 단위테스트 수정

- service도입으로 변경된 repository에 대해 테스트도 수정하여 진행하였습니다.

tncks93 committed:
[#6] Feat: OwnerArgumentResolver 구현

- ArgumentResolver를 통해 Controller에서 Owner객체를 파라미터로 받도록 구현
- 공통 응답 포맷을 구현
- controller에서 repository가 아닌 service를 의존하도록 수정

tncks93 committed:
[#6] Feat: OwnerArgumentResolver 구현

- ArgumentResolver를 통해 Controller에서 Owner객체를 파라미터로 받도록 구현
- 공통 응답 포맷을 구현
- controller에서 repository가 아닌 service를 의존하도록 수정

f-lab-lyan commented on api/src/main/java/com/delivery_service/owners/repository/OwnerRepository.java:
```diff
@@ -0,0 +1,16 @@
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class OwnerRepository {
```
OwnerRepository에 대한 세부 구현을 진행하지 않을 예정이시라면, 그냥 owner정보를 여기에 몇 개 추가해 놓고, 사용하면 되지 않을까요? 나중에 디비를 쓰게 되면, 그 때도, DB에 정보를 미리 넣어놓고 사용할 수 있을 것 같습니다.

tncks93 replied:
네 적용했습니다.

f-lab-lyan reviewed (changes requested):
Repository 테스트는 있는데, 정작 서비스 테스트는 없는 것 같네요. :-) 서비스 테스트도 추가해 주시겠어요? 나중에 로직이 좀 더 추가되면, 더 복잡한 테스트가 필요하겠지만, 일단은 간단한 테스트라도 추가해놓으면 좋을 것 같습니다.

tncks93 committed:
[#6] Style: 로그 레벨 및 오타 수정

- log 레벨 info -> debug로 변경
- 로그 문자 일부 오타 수정

tncks93 committed:
Merge pull request #8 from tncks93/feature/6-add-argument-resolver

[#6] OwnerArgumentResolver 및 공통 응답 포맷 구현

tncks93 committed:
[#6] Refactor: 공통 응답 포맷 수정 및 적용

- 사용성 편의를 위해 응답 결과에 따른 static 메소드 구현
- controller에 적용

tncks93 committed:
[#6] Test: OwnerShopService 단위 테스트 작성

- Owner의 Shop 추가, 수정, 조회, OPEN/CLOSE 상태 변경의 단위 테스트 작성

tncks93 commented:
서비스 단위 테스트 작성했습니다.

tncks93 committed:
[#6] Refactor: OwnerArgumentResolver, 공통 응답 포맷 코드 수정

- Session이 아닌 Header를 통해 Owner를 식별하고, repository를 통해 Owner를 가져오도록 코드 수정
- OwnerRespository에서 dummy로 생성된 Owner객체 사용
- CommonResponse의 생성자 문제로 테스트시 문제 발생으로 @NoArgsConstructor 추가

tncks93 committed:
[#6] Refactor: OwnerShopController 통합테스트 수정

- session 사용 부분 삭제
- header를 통해 owner 식별 정보 전달
- api별로 테스트 코드 분리

tncks93 commented:
제가 실수로 하위 브랜치를 merge해 버렸습니다. revert시 어떻게 될지 확신이 없어 진행하지 않았습니다. git에 대해 공부해 보겠습니다.

tncks93 commented:
현재 pr과 다른 브랜치에서의 pr에서 제안해주신 부분 추가,수정해서 commit 했습니다.

tncks93 committed:
[#6] Style: 오타 수정

- 일부 오타 수정
- 누락된 어노테이션 추가
- 공백 라인 일부 제거

tncks93 committed:
[#6] Chore: application.properties 파일 추가

- 프로젝트 로그 레벨을 'debug'로 변경

f-lab-lyan commented on api/src/test/java/com/delivery_service/owners/service/OwnerShopServiceTest.java:
```diff
@@ -0,0 +1,155 @@
+        registerDto.setCategory("패스트푸드");
+        Shop shopToSave = registerDto.convertToEntity();
+        Owner owner = new Owner(1);
+        when(ownerShopRepository.save(any(Shop.class))).thenAnswer(invocation -> {
```
응답이 정해져있는 경우에는 정해진 응답을 그냥 리턴하는 것이 더 명확하지 않을까 싶네요.

f-lab-lyan commented on api/src/main/java/com/delivery_service/owners/resolver/OwnerArgumentResolver.java:
```diff
@@ -33,20 +37,32 @@ public boolean supportsParameter(MethodParameter parameter) {
-        Owner owner = ownerRepository.findById(ownerId);
-        log.debug("owner = {}",owner);
-        return owner;
+            byte[] base64Credentials = Base64.getDecoder().decode(authorization.split(" ")[1]);
```
base64로 하면, 보안에 취약할 것 같아요. 다른 방법이 필요할 것 같습니다.

f-lab-lyan commented on api/src/main/resources/application.properties:
```diff
@@ -0,0 +1,2 @@
+
+logging.level.com.delivery_service=debug
```
logback.xml을 도입해 보는 것이 어떨까 싶네요.

f-lab-lyan commented on api/src/test/java/com/delivery_service/owners/service/OwnerShopServiceTest.java:
```diff
@@ -0,0 +1,155 @@
+@ExtendWith(MockitoExtension.class)
+class OwnerShopServiceTest {
+    @Mock
+    OwnerRepository ownerRepository;
```
OwnerService에서도 OwnerRepository를 불러서 쓰지 않을까요?

f-lab-lyan replied:
https://www.baeldung.com/circular-dependencies-in-spring

f-lab-lyan replied:
https://ko.wikipedia.org/wiki/%ED%8D%BC%EC%82%AC%EB%93%9C_%ED%8C%A8%ED%84%B4
